app-modules/multifactor-authentication/src/Services/MultifactorService.php

```diff
@@ -0,0 +1,111 @@
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\MultifactorAuthentication\Services;
+
+use App\Models\User;
+use BaconQrCode\Writer;
+use Filament\Facades\Filament;
+use Illuminate\Cache\Repository;
+use PragmaRX\Google2FA\Google2FA;
+use BaconQrCode\Renderer\Color\Rgb;
+use BaconQrCode\Renderer\ImageRenderer;
+use BaconQrCode\Renderer\RendererStyle\Fill;
+use BaconQrCode\Renderer\Image\SvgImageBackEnd;
+use BaconQrCode\Renderer\RendererStyle\RendererStyle;
+
+class MultifactorService
+{
+    public function __construct(
+        protected Google2FA $engine,
+        protected ?Repository $cache = null
+    ) {}
+
+    public static function make(): static
+    {
+        return app(static::class);
+    }
+
+    public function getEngine()
+    {
+        return $this->engine;
+    }
+
+    public function generateSecretKey()
+    {
+        return $this->engine->generateSecretKey();
+    }
+
+    public function getMultifactorQrCodeSvg(string $url)
+    {
+        $svg = (new Writer(
+            new ImageRenderer(
+                new RendererStyle(150, 1, null, null, Fill::uniformColor(new Rgb(255, 255, 255), new Rgb(45, 55, 72))),
+                new SvgImageBackEnd()
+            )
+        ))->writeString($url);
+
+        return trim(substr($svg, strpos($svg, "\n") + 1));
+    }
+
+    public function getQrCodeUrl($companyName, $companyEmail, $secret)
+    {
+        return $this->engine->getQRCodeUrl($companyName, $companyEmail, $secret);
+    }
+
+    public function verify(string $code, ?User $user = null)
+    {
+        if (is_null($user)) {
+            $user = Filament::auth()->user();
```

Comment:
Why use the filament user and not the normal auth user?

Reply:
This is build to work with the Filament panel auth